### PR TITLE
[improve](cloud) Optimize meta service proxy retry strategy

### PR DESCRIPTION
**Pull request by w41ter**
## Proposed changes

Issue Number: close #xxx

1. use exponential backoff to reduce retrying costs。
2. KV_TXN_TOO_OLD only retries once because it is expensive

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: cloud/src/meta-service/meta_service.h [hunk 1]
```diff
@@ -616,39 +613,53 @@ class MetaServiceProxy final : public MetaService {
...
+                code != MetaServiceCode::KV_TXN_TOO_OLD &&
                 (!config::enable_retry_txn_conflict || code != MetaServiceCode::KV_TXN_CONFLICT)) {
                 return;
             }
 
             TEST_SYNC_POINT("MetaServiceProxy::call_impl:2");
-            if (--retry_times < 0) {
+            if (retry_times == 0) {
+                // the first retry, add random drift.
+                std::uniform_int_distribution<uint64_t> u(
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.h:632]: the RNG can be made as thread local?

File: cloud/test/fdb_injection_test.cpp [hunk 2]
```diff
@@ -67,6 +67,7 @@ int main(int argc, char** argv) {
 
     cloud::config::enable_txn_store_retry = true;
     cloud::config::txn_store_retry_times = 100;
+    cloud::config::txn_store_retry_base_intervals_ms = 1;
```
Comment (gavinchou) [inline cloud/test/fdb_injection_test.cpp:70]: 1ms may be too short to prevent an avalanche

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: cloud/test/fdb_injection_test.cpp (hunk 2 above)
Comment (w41ter) [reply cloud/test/fdb_injection_test.cpp:70]: This test is mainly used to test whether ms proxy successfully retries these errors.

**Comment by w41ter** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**